FTBFS

Lua: 5.2.2

In file included from /home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_common.c:25:0:
/home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_common.h:24:28: error: expected declaration specifiers or '...' before '(' token
 luaL_register (lua_State *L, const gchar *name, const struct luaL_reg *methods)
                            ^
/home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_common.h:24:48: error: expected declaration specifiers or '...' before '(' token
 luaL_register (lua_State *L, const gchar *name, const struct luaL_reg *methods)
                                                ^
/home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_common.h:24:81: error: expected declaration specifiers or '...' before numeric constant
 luaL_register (lua_State *L, const gchar *name, const struct luaL_reg *methods)
                                                                                 ^
src/lua/CMakeFiles/rspamd-lua.dir/build.make:57: recipe for target 'src/lua/CMakeFiles/rspamd-lua.dir/lua_common.c.o' failed
make[2]: *** [src/lua/CMakeFiles/rspamd-lua.dir/lua_common.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_message.c:26:0:
/home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_common.h:24:28: error: expected declaration specifiers or '...' before '(' token
 luaL_register (lua_State *L, const gchar *name, const struct luaL_reg *methods)
                            ^
/home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_common.h:24:48: error: expected declaration specifiers or '...' before '(' token
 luaL_register (lua_State *L, const gchar *name, const struct luaL_reg *methods)
                                                ^
/home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_common.h:24:81: error: expected declaration specifiers or '...' before numeric constant
 luaL_register (lua_State *L, const gchar *name, const struct luaL_reg *methods)
                                                                                 ^
src/lua/CMakeFiles/rspamd-lua.dir/build.make:103: recipe for target 'src/lua/CMakeFiles/rspamd-lua.dir/lua_message.c.o' failed
make[2]: *** [src/lua/CMakeFiles/rspamd-lua.dir/lua_message.c.o] Error 1
In file included from /home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_task.c:26:0:
/home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_common.h:24:28: error: expected declaration specifiers or '...' before '(' token
 luaL_register (lua_State *L, const gchar *name, const struct luaL_reg *methods)
                            ^
/home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_common.h:24:48: error: expected declaration specifiers or '...' before '(' token
 luaL_register (lua_State *L, const gchar *name, const struct luaL_reg *methods)
                                                ^
/home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9/src/lua/lua_common.h:24:81: error: expected declaration specifiers or '...' before numeric constant
 luaL_register (lua_State *L, const gchar *name, const struct luaL_reg *methods)
                                                                                 ^
src/lua/CMakeFiles/rspamd-lua.dir/build.make:80: recipe for target 'src/lua/CMakeFiles/rspamd-lua.dir/lua_task.c.o' failed
make[2]: *** [src/lua/CMakeFiles/rspamd-lua.dir/lua_task.c.o] Error 1
make[2]: Leaving directory '/home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9'
CMakeFiles/Makefile2:242: recipe for target 'src/lua/CMakeFiles/rspamd-lua.dir/all' failed
make[1]: *** [src/lua/CMakeFiles/rspamd-lua.dir/all] Error 2
make[1]: Leaving directory '/home/rpmaker/rpmbuild/BUILD/rspamd-0.6.9'
Makefile:120: recipe for target 'all' failed
make: *** [all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.28sXjs (%build)

<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/3624985-ftbfs?utm_campaign=plugin&utm_content=tracker%2F320726&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F320726&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 #
-# Rspamd - rambler antispam system
+# Rspamd - rapid antispam system
 #
 # Cmake configuration file
 #

--- a/centos/rspamd.spec
+++ b/centos/rspamd.spec
@@ -30,35 +30,27 @@ License:        BSD2c
 %endif
 URL:            https://rspamd.com
 BuildRoot:      %{_tmppath}/%{name}-%{version}-%{release}
-%if 0%{?suse_version}
-BuildRequires:  cmake,glib2-devel,gmime-devel,libevent-devel,openssl-devel,lua-devel,pcre-devel,perl,sqlite3-devel
-%endif
-%if 0%{?el6}
-BuildRequires:  cmake,glib2-devel,gmime-devel,libevent-devel,openssl-devel,lua-devel,pcre-devel,perl,sqlite-devel
-%else
-BuildRequires:  cmake,glib2-devel,gmime-devel,libevent-devel,openssl-devel,lua-devel,pcre-devel,perl,sqlite-devel,systemd
-%endif
-%if 0%{?el6}
-Requires:       lua, logrotate
-%else
-Requires:       lua
-%endif
-# for /user/sbin/useradd
-%if 0%{?suse_version}
-Requires(pre):  shadow
-%else
-Requires(pre):  shadow-utils
-%endif
-%if 0%{?suse_version} || 0%{?fedora} || 0%{?el7}
-Requires(pre): systemd
+BuildRequires:  cmake,glib2-devel,gmime-devel,libevent-devel,openssl-devel,lua-devel,pcre-devel,perl
+%if 0%{?suse_version} || 0%{?el7} || 0%{?fedora}
+BuildRequires:  systemd
+Requires(pre):  systemd
 Requires(post): systemd
 Requires(preun): systemd
 Requires(postun): systemd
+%endif
+%if 0%{?suse_version}
+BuildRequires:  sqlite3-devel
+Requires(pre):  shadow
 %else
+BuildRequires:  sqlite-devel
+Requires(pre):  shadow-utils
+%endif
+Requires:       lua
+%if 0%{?el6}
+Requires:       logrotate
 Requires(post): chkconfig
-# for /sbin/service
-Requires(preun):        chkconfig, initscripts
-Requires(postun):       initscripts
+Requires(preun): chkconfig, initscripts
+Requires(postun): initscripts
 %endif
 
 Source0:        https://rspamd.com/downloads/%{name}-%{version}.tar.xz

